services/localizely/localizely-base.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService } = require('..')
+
+const keywords = [
+  'l10n',
+  'i18n',
+  'localization',
+  'internationalization',
+  'translation',
+  'translations',
+]
+
+const documentation = `
+  <p>
+    The read-only API token from the Localizely account is required to fetch necessary data.
+    <br/>
+    You can find more details regarding API tokens under <a href="https://app.localizely.com/account" target="_blank">My profile</a> page.
+  </p>
+  `
+
+const schema = Joi.object({
+  strings: Joi.number().required(),
+  reviewedProgress: Joi.number().required(),
+  languages: Joi.array()
+    .items(
+      Joi.object({
+        langCode: Joi.string().required(),
+        langName: Joi.string().required(),
+        strings: Joi.number().required(),
+        reviewed: Joi.number().required(),
+        reviewedProgress: Joi.number().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({ token: Joi.string() }).required()
+
+class BaseLocalizelyService extends BaseJsonService {
+  async fetch({ projectId, branch, apiToken }) {
+    return this._requestJson({
+      schema,
+      url: `https://api.localizely.com/v1/projects/${projectId}/status`,
+      options: {
+        qs: { branch },
+        headers: { Accept: 'application/json', 'X-Api-Token': apiToken },
```

Comment:
Thanks for the feedback!

Maybe I'm missing something here, but according to the `_requestJson` implementation and the [querying-an-api doc](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md#43-querying-an-api), headers will be overwritten in case they are provided.

```
// base-json.js
async _requestJson({ schema, url, options = {}, errorMessages = {} }) {
    const mergedOptions = {
      ...{ headers: { Accept: 'application/json' } },
      ...options,
    }
    const { buffer } = await this._request({
      url,
      options: mergedOptions,
      errorMessages,
    })
    const json = this._parseJson(buffer)
    return this.constructor._validate(json, schema)
  }
```

Reply:
You're correct that the explicit json mode based that ultimately gets passed to our http client lib (request) does have the json mode wiped out due to the object merging.

However, it's not necessary to specify the header here because the server in this case (Localizely API) will only ever return a json response, so there's no negotiation needed. The base infra will also [handle the parsing of the response](https://github.com/badges/shields/blob/bf98260a4c41e69844a1fcba840a45c5e83d6aac/core/base-service/json.js#L8-L24) and handle any errors accordingly. 

The API doc you linked also states immediately after the referenced text:

> There is no need to specify json, as the JSON parsing is handled by _requestJson()

It's not a big deal one way or the other, so don't feel like you have to remove, just noting it's not explicitly necessary to include it. Some of our other json-based classes with additional headers include Accept anyway and others do not

Reply:
Applied suggestion for removing the `Accept` header within [9028aa](https://github.com/badges/shields/pull/5472/commits/9028aa1857e4887071e6b4a9f948df597ce44612) commit.